タワーオブジェクトまわりの不具合修正

tr-t-mitsumaru opened the pull request:
前回発見された不具合を一通り修正した。
一番下のオブジェクトの位置がズレていく不具合を修正し、プレイヤーのUnityEventに対応させた。
クラスの名前も一部変更した。
移動による落下処理をLerpではなく、positionを直接変更するようにした。

【確認ファイル】
MoveControllerBase.cs
MoveControllerList.cs
ObjectFallingController.cs
TowerObjectActionCaller.cs
TowerObjectSpawner.cs
MochiMuveController.cs
RabbitMoveController.cs
RabbitData.cs
DrawObjectName.cs
（オブジェクトの名前をテキストで表示するデバッグ用プログラムです）

tr-t-mitsumaru committed:
タワーオブジェクトまわりの不具合修正

前回発見された不具合を一通り修正した。
一番下のオブジェクトの位置がズレていく不具合を修正し、プレイヤーのUnityEventに対応させた。
クラスの名前も一部変更した。
移動による落下処理をLerpではなく、positionを直接変更するようにした。

【確認ファイル】
MoveControllerBase.cs
MoveControllerList.cs
ObjectFallingController.cs
TowerObjectActionCaller.cs
TowerObjectSpawner.cs
MochiMuveController.cs
RabbitMoveController.cs
RabbitData.cs
DrawObjectName.cs
（オブジェクトの名前をテキストで表示するデバッグ用プログラムです）

tr-t-mitsumaru committed:
エラー修正とコメント追加

リスト関連の処理でnullを参照しようとしてエラーが出ていたので修正した。
コメントが不足していた箇所があったので追加した。

tr-k-oishi commented on Assets/Scripts/TowerDatas/Common/TowerFlyController.cs:
```diff
@@ -1,53 +1,19 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
-
```
エラー避けならコメントは書いといたほうがいいかも

tr-k-oishi commented on Assets/Scripts/TowerDatas/Common/TowerObjectActionCaller.cs:
```diff
@@ -17,102 +19,70 @@ public class TowerObjectActionCaller : MonoBehaviour
-        // デバッグ用のプレイヤーアクション
-        // A : パンチ
-        // S : 救出
+        // 一番下のオブジェクトを取得
```
同じの３こ続いてるから関数化したほうがいいかも

tr-k-oishi commented on Assets/Scripts/TowerDatas/Common/TowerObjectSpawner.cs:
```diff
@@ -12,49 +12,57 @@ public class TowerObjectSpawner : MonoBehaviour
     // 現在のモチのスキン
     SettingData.SkinType mochiSkinType = SettingData.SkinType.NormalMochi;
 
+    List<Transform> spawnedMochiList = new List<Transform>();
```
細かいかもしれんけどコメント書いとったほうがええかも

tr-k-oishi commented on Assets/Scripts/TowerDatas/Common/TowerObjectSpawner.cs:
```diff
@@ -66,47 +74,46 @@ public void Spawn(int spawnNum)
+                spawnedObject = mochiSpawnPool.Spawn(mochiSkinType.ToString(), spawnPos, Quaternion.identity);
                 // 生成したオブジェクトをオンにする
                 // NOTE : 何故か自動でオンになってくれないときがあるため
                 spawnedObject.gameObject.SetActive(true);
```
spawnedObject.gameObject.SetActive(true);
stackedObjects.Add(spawnedObject.transform);
spawnedMochiList.Add(spawnedObject);
は両方とも一緒なのでif文の外で大丈夫だと思う

tr-t-mitsumaru committed:
指摘された箇所を修正

同じような処理を関数化して、一つにまとめた。
不足していたコメントを追加した。

【確認ファイル】
TowerObjectActionCaller.cs
TowerObjectSpawner.cs

tr-t-mitsumaru committed:
不足コメント追加

追加したコメントが保存できておらず、反映されていなかったので修正

tr-k-oishi reviewed (approved):
LGTM

tricore-as04 reviewed (approved):
LGTM
